src/core/core.datasetController.js

```diff
@@ -234,6 +235,47 @@ helpers.extend(DatasetController.prototype, {
 		me.resyncElements();
 	},
 
+	/**
+	 * Private functionality to be called before every update
+	 * @private
+	 */
+	_update: function() {
+		var me = this;
+		me._config = me._configure();
+		me.update.apply(me, arguments);
+	},
+
+	/**
+	 * Returns the merged user-supplied and default dataset-level options
+	 * @private
+	 */
+	_configure: function() {
```

Comment:
As discussed on Slack, we don't need to merge dataset with defaults but instead fallback separately to be able to handle "user over defaults" correctly. So this method is not anymore useful.

Reply:
Done

Reply:
I'm not sure how this was resolved.

Reply:
A previous version of the PR was trying to have user-supplied values always take precedence over defaults. I believe Simon's comment was suggesting to remove this method just in that case to handle "user over defaults" correctly. After writing the code to go down that path I figured out that it had issues and we shouldn't try to do it. I talked to Simon in the #dev channel and he suggested it would be better to always have element options including defaults take precedence over dataset options in v2. I agreed and changed the code to do that and so I think this comment is no longer relevant. I believe that this is the preferred method of implementing the current solution as it had been suggested by Simon prior to trying to go the "user over defaults" route.